spec/models/hw_info_spec.rb

```diff
@@ -0,0 +1,45 @@
+require 'spec_helper'
+require 'rails_helper'
+
+describe HwInfo do
+  subject(:hw_info) { create(:system, :with_hw_info).hw_info }
+
+  let(:valid_uuid) { 'aaaaaaaa-aaaa-4aaa-9aaa-aaaaaaaaaaaa' }
+
+  it { is_expected.to belong_to :system }
+  it { is_expected.to validate_presence_of :system }
+
+  it 'enforces uniqueness' do
+    expect(hw_info).to validate_uniqueness_of(:system)
+  end
+
+  describe '.uuid' do
+    it { is_expected.to allow_value(nil).for(:uuid) }
+    it { is_expected.to allow_value('xyzzy').for(:uuid) }
+
+    it 'forces invalid uuid to nil' do
+      hw_info.uuid = 'xyzzy'
+      hw_info.save!
+
+      expect(hw_info.uuid).to be nil
+    end
+
+    it_behaves_like 'model with UUID format validation and nil forcing', :uuid
```

Comment:
Is this actually testing anything worthwhile? It seems that the shared example will pass with anything and you're not checking if the model is nil at the end.

Also, this is only used once, is a shared example really needed here?

Reply:
well I think - maybe this is not needed to use shared example. it is already tested by other specs in this file